src/meta/src/stream/scheduler.rs

```diff
@@ -92,6 +92,32 @@ impl ScheduledLocations {
             })
             .collect::<Vec<_>>()
     }
+
+    /// Find a placement location that is on the same node of given actor ids.
+    pub fn ensure_placement_rule(&self, actor_ids: &[ActorId]) -> Result<ParallelUnit> {
```

Comment:
Can we avoid introducing the concept 'placement rule'? It sounds too heavy to me.

```suggestion
    pub fn schedule_colocate_with(&self, actor_ids: &[ActorId]) -> Result<ParallelUnit> {
```